### ISSUE TITLE
owner / group of azurerm_storage_data_lake_gen2_filesystem can't be $superuser despite API supporting it

### ISSUE THREAD

**Issue by flobeier** [hashicorp/terraform-provider-azurerm#16138]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
* Terraform v1.1.3
* azurerm 2.99.0

### Affected Resource(s)

* `azurerm_storage_data_lake_gen2_filesystem`
* `azurerm_storage_data_lake_gen2_path`

### Terraform Configuration Files

```hcl
resource "azurerm_storage_data_lake_gen2_filesystem" "example" {
  name               = "example"
  storage_account_id = azurerm_storage_account.example.id
  
  owner = "$superuser"
  group = "$superuser"
}
```

### Expected Behaviour

Terraform sees the code as valid and applies it because the API supports it (verified with `az storage fs access set --group '$superuser' ...`).

### Actual Behaviour

```log
Error: expected "owner" to be a valid UUID, got $superuser

Error: expected "group" to be a valid UUID, got $superuser
```
I'm not sure though if this behavior intentional, I'm just trying to replicate infrastructure set up by Bash scripts in Terraform code 1:1 without changing anything.

### Steps to Reproduce

1. `terraform apply`

**Comment by flobeier** [comment]
@catriona-m can you update the `azurerm_storage_data_lake_gen2_path` resource too?

**Comment by catriona-m** [comment]
Sorry @flobeier - I missed that resource when I did this initially! `azurerm_storage_data_lake_gen2_path` is fixed here: #16370

Thanks!

**Comment by flobeier** [comment]
@catriona-m thanks a lot! :+1:
